package/base-files/files/sbin/sysupgrade

```diff
@@ -257,18 +257,21 @@ do_save_conffiles() {
 
 	v "Saving config files..."
 	[ "$VERBOSE" -gt 1 ] && TAR_V="v" || TAR_V=""
-	tar c${TAR_V}zf "$conf_tar" -T "$CONFFILES" 2>/dev/null
-	if [ "$?" -ne 0 ]; then
+	sed -i -e 's,^/,,' "$CONFFILES"
```

Comment:
I love writing scripts that use only pipes without saving temporary data to disks. We could remove the '/' in both add_overlayfiles() and add_conffiles(), and reintroduce it when the user requests sysupgrade to list files (-l). However, I believe this approach is more complex than rewriting the file immediately where the slash must be removed.

Until someone fails to generate a backup because two lists of files temporary coexist in the /tmp directory, I suggest sticking with the simpler solution. I think the space for two lists was already required (and freed) during add_overlayfiles(), for example. Sysupgrade heavily relies on temporary files, and the ash compatibility makes it difficult to replace them with pipes.

Reply:
`$CONFFILES` is in `/tmp/` and we call that `sed` once per `sysupgrade`, I really wouldn't be concerned too much about its cost here, especially since it's cleaner this way